### PR TITLE
middleware: made 404 info more useful 

### LINKED ISSUE #7711: Make 404 more useful when possible
### Describe the problem

For example, if you try go to a non-existent language of a component, the page will only suggest to go home or to all projects, while it would be better if it linked to the component, or suggested adding a language that you tried to visit. Example: https://hosted.weblate.org/projects/keepassxc/keepassxc-browser-dev/fil/

image

### Describe the solution you'd like

Add useful suggestions depending on the context.

### Describe alternatives you've considered

_No response_

### Screenshots

_No response_

### Additional context

https://github.com/WeblateOrg/weblate/blob/1804a951c8ae9199d23099a8ec322134ee5b45bc/weblate/urls.py#L136-L140

https://github.com/WeblateOrg/weblate/blob/329dabf59991ed2ee1b44f331929da41d8031c98/weblate/templates/404.html

### PR DESCRIPTION
**Pull request by ParthS007**
## Proposed changes

closes #7711 

## Checklist

- [x] Lint and unit tests pass locally with my changes.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] I have added documentation to describe my feature.
- [x] I have squashed my commits into logic units.
- [x] I have described the changes in the commit messages.

## Other information

Screenshot 2022-07-20 at 01 06 35

### CODE REVIEW

File: weblate/middleware.py [hunk 1]
```diff
@@ -180,6 +184,21 @@ def process_exception(self, request, exception):
...
+                new_lang = kwargs.get("new_language")
+                if new_lang not in project.languages:
+                    messages.add_message(
+                        request,
+                        messages.INFO,
+                        _(f"{new_lang} is currently not available, but can be added."),
+                    )
+                    return redirect(
+                        reverse(
+                            "component", args=[kwargs["project"], kwargs["component"]]
```
Comment (nijel) [inline weblate/middleware.py:197]: The code is reachable when component is not set....

Comment (ParthS007) [reply weblate/middleware.py]: Thanks for the catch. Updated.

I am not sure how weblate shows info messages to the User. I used messages for this. let me know if we have something already.

Comment (nijel) [reply weblate/middleware.py]: Using messages is the correct way to go.

**Comment by nijel** [comment]
Merged, thanks for your contribution!

**Pull request merged**